packages/openapi-code-generator/src/typescript/common/schema-builders/joi-schema-builder.spec.ts

```diff
@@ -1068,6 +1073,181 @@ describe.each(testVersions)(
+      })
+
+      it("can union an intersected object and string", async () => {
+        const {code, execute} = await getActualFromModel(
+          irModelObject({
+            anyOf: [
+              irModelString(),
+              irModelObject({
+                allOf: [
+                  irModelObject({
+                    properties: {foo: irModelString()},
+                    required: ["foo"],
+                  }),
+                  irModelObject({
+                    properties: {bar: irModelString()},
+                    required: ["bar"],
+                  }),
+                ],
+              }),
+            ],
+          }),
+        )
+
+        expect(code).toMatchInlineSnapshot(`
+          "const x = joi
+            .alternatives()
+            .try(
+              joi.string().required(),
+              joi
+                .object()
+                .keys({ foo: joi.string().required() })
+                .options({ stripUnknown: true })
+                .required()
+                .concat(
+                  joi
+                    .object()
+                    .keys({ bar: joi.string().required() })
+                    .options({ stripUnknown: true })
+                    .required(),
+                )
+                .required(),
+            )
+            .required()"
+        `)
+
+        await expect(execute("some string")).resolves.toEqual("some string")
+        await expect(execute({foo: "bla", bar: "foobar"})).resolves.toEqual({
+          foo: "bla",
+          bar: "foobar",
+        })
+        await expect(execute({foo: "bla"})).rejects.toThrow('"bar" is required')
+      })
+    })
+
+    describe("intersections", () => {
+      it("can intersect objects", async () => {
+        const {code, execute} = await getActualFromModel(
+          irModelObject({
+            allOf: [
+              irModelObject({
+                properties: {foo: irModelString()},
+                required: ["foo"],
+              }),
+              irModelObject({
+                properties: {bar: irModelString()},
+                required: ["bar"],
+              }),
+            ],
+          }),
+        )
+
+        expect(code).toMatchInlineSnapshot(`
+          "const x = joi
+            .object()
+            .keys({ foo: joi.string().required() })
+            .options({ stripUnknown: true })
+            .required()
+            .concat(
+              joi
+                .object()
+                .keys({ bar: joi.string().required() })
+                .options({ stripUnknown: true })
+                .required(),
+            )
+            .required()"
+        `)
+
+        await expect(execute({foo: "bla", bar: "foobar"})).resolves.toEqual({
+          foo: "bla",
+          bar: "foobar",
+        })
+        await expect(execute({foo: "bla"})).rejects.toThrow('"bar" is required')
+      })
+
+      // TODO: https://github.com/hapijs/joi/issues/3057
```

Comment:
Skipping this test so that I can merge the fix for the `zod` side without blocking on the `joi` builder. Raised an issue https://github.com/hapijs/joi/issues/3057 to hopefully get some feedback on any existing ways to solve this